ocaml/typing/typecore.ml

```diff
@@ -5959,7 +5973,7 @@ and type_expect_
       }
   | Pexp_send (e, {txt=met}) ->
       submode ~loc ~env Mode.Value.legacy expected_mode;
-      let pm = position_and_mode env expected_mode sexp in
+      let pm = position_and_mode env expected_mode sexp ~maybe_ident:None in
```

Comment:
I agree we're not worried about changing how locals interact with method dispatch and constructors, so we can keep the default behavior of always producing tail calls. I wonder if we should make it more explicit that's what we're doing, especially since it might become easier to accidentally change the behavior here when we start using more complicated heuristics.

Reply:
Ack; improved in next feature.